### PR TITLE
Accumulo 1480 Added namespace filtering to tables

### PR DESCRIPTION
**Pull request by lstav**
Can now filter namespaces on the table list. 

Back end implementation is done, and I would appreciate the feedback on the Front end. 

Front end can be seen on the "Master Server" and "Tables" pages.

### CODE REVIEW

**Comment by ctubbsii** [comment]
Mentioning ACCUMULO-1480 so it links to the JIRA issue.

**Comment by lstav** [comment]
Pushed a commit to fix the checkstyle error I was getting on the build.

File: server/monitor/src/main/java/org/apache/accumulo/monitor/util/Table.java [hunk 1]
```diff
@@ -143,7 +146,24 @@ public synchronized void generate(HttpServletRequest req, StringBuilder sb) {
...
+
+    if (namespaces != null) {
+      sb.append("<div id=\"filters\">\n");
+      String namespace = BasicServlet.getCookieValue(req, "namespaceDropdown." + BasicServlet.encode(page) + "." + BasicServlet.encode(tableName) + "."
+          + "selected");
+      if (namespace == null) {
+        namespace = "*";
+      }
+      sb.append("<div class='left show'><dl>\n");
+      doDropdownMenu(redir, page, tableName, sb, namespaces, namespace);
```
Comment (mikewalch) [inline server/monitor/src/main/java/org/apache/accumulo/monitor/util/Table.java]: Could add header to menu like "Table namespaces" or "Namespaces"

File: server/monitor/src/main/java/org/apache/accumulo/monitor/util/Table.java [hunk 2]
```diff
@@ -227,4 +247,25 @@ private static void row(StringBuilder sb, boolean highlight, ArrayList<TableColu
     sb.append("</tr>\n");
   }
 
+  private static void doDropdownMenu(String redir, String page, String tableName, StringBuilder sb, SortedMap<String,String> namespaces, String namespace) {
+
+    String namespaceUrl = String.format("/op?action=namespace&redir=%s&page=%s&table=%s&selected=", redir, page, tableName);
+
+    sb.append("<ul id=\"namespaces\">\n");
+    sb.append("<li><a ").append(namespace.equals("*") ? "class=\"active\" " : "").append("href=\"").append(namespaceUrl).append("*\">*</a></li>");
```
Comment (mikewalch) [inline server/monitor/src/main/java/org/apache/accumulo/monitor/util/Table.java]: `*` is a little confusing on its own.  Could add something about this being all tables

File: server/monitor/src/main/java/org/apache/accumulo/monitor/util/Table.java [hunk 3]
```diff
@@ -227,4 +247,25 @@ private static void row(StringBuilder sb, boolean highlight, ArrayList<TableColu
...
+    for (String key : namespaces.keySet()) {
+      if (key.equals("")) {
+        sb.append("<li><a ").append(namespace.equals("-") ? "class=\"active\" " : "").append("href=\"").append(namespaceUrl).append("-\">-</a></li>");
```
Comment (mikewalch) [inline server/monitor/src/main/java/org/apache/accumulo/monitor/util/Table.java]: Same as above. `-` is a little confusing.  Could add text describing this option as default tables.

File: server/monitor/src/main/java/org/apache/accumulo/monitor/util/Table.java (hunk 1 above)
Comment (lstav) [inline server/monitor/src/main/java/org/apache/accumulo/monitor/util/Table.java]: Will add a header "Namespaces" that uses a different font than the rest of the list.

File: server/monitor/src/main/java/org/apache/accumulo/monitor/util/Table.java (hunk 2 above)
Comment (lstav) [inline server/monitor/src/main/java/org/apache/accumulo/monitor/util/Table.java]: Will change it to be "* (All Tables)"

File: server/monitor/src/main/java/org/apache/accumulo/monitor/util/Table.java (hunk 3 above)
Comment (lstav) [inline server/monitor/src/main/java/org/apache/accumulo/monitor/util/Table.java]: Will change this to be "- (DEFAULT)"

**Comment by lstav** [comment]
Here are screenshots of the Tables Page and the Master Server Page which shows the tables.

Tables Page
tables_page

Master Server Page
master_page

**Comment by mikewalch** [comment]
The changes that you made in response to my review look good.

**Comment by lstav** [comment]
Thanks!

**Comment by lstav** [comment]
I'm OK if you merge it in now. I'll work on the improvements on top of the changes I made here.

**Pull request merged**